client_plugin/drivers/shared/dockerops/dockerops.go

```diff
@@ -142,13 +175,165 @@ func (d *DockerOps) VolumeRemove(volName string) error {
 }
 
 // StartSMBServer - Start SMB server
-func (d *DockerOps) StartSMBServer(volName string) bool {
-	log.Errorf("startSMBServer to be implemented")
-	return true
+func (d *DockerOps) StartSMBServer(volName string) (int, string, bool) {
+	var service swarm.ServiceSpec
+	var options dockerTypes.ServiceCreateOptions
+
+	// Name of the service
+	service.Name = volName + serviceNamePostfix
+	// The Docker image to run in this service
+	service.TaskTemplate.ContainerSpec.Image = sambaImageName
+
+	/* Args which will be passed to the service. These options are
+	   * used by the Samba container, not Docker API.
+	   * -s: Share related info: Name of the share,
+	                             Path in the Samba container that will be shared,
+	                             Browsable (yes),
+	                             Read only (no),
+	                             Guest access allowed by default (no),
+	                             Which users can access (all),
+	                             Which users are admins? (root)
+	                             Writelist: If RO, who can write on the share (root)
+	   * -u: Username and Password
+	*/
+	containerArgs := []string{"-s",
+		fileShareName + ";/mount;yes;no;no;all;" +
+			SambaUsername + ";" + SambaUsername,
+		"-u",
+		SambaUsername + ";" + SambaPassword}
+	service.TaskTemplate.ContainerSpec.Args = containerArgs
+
+	// Mount a volume on service containers at mount point "/mount"
+	var mountInfo []swarm.Mount
+	mountInfo = append(mountInfo, swarm.Mount{
+		Type:   swarm.MountType("volume"),
+		Source: internalVolumePrefix + volName,
+		Target: "/mount"})
+	service.TaskTemplate.ContainerSpec.Mounts = mountInfo
+
+	// How many containers of this service should be running at a time?
+	// Service mode can be Replicated or Global
+	var uintContainerNum uint64
+	uintContainerNum = 1
+	numContainers := swarm.ReplicatedService{Replicas: &uintContainerNum}
+	service.Mode = swarm.ServiceMode{Replicated: &numContainers}
+
+	/* Ports that the service wants to expose
+	   * Protocol: Samba operates on TCP
+	   * TargetPort: The port within the container that we wish to expose.
+	                 Port on host VM will get self assigned.
+	*/
+	var exposedPorts []swarm.PortConfig
+	exposedPorts = append(exposedPorts, swarm.PortConfig{
+		Protocol:   swarm.PortConfigProtocolTCP,
+		TargetPort: defaultSambaPort,
+	})
+	// service.EndpointSpec is an input for service create.
+	// It carries the previous data structure as well as Mode.
+	// ResolutionModeVIP indicates that networking uses virtual IPs.
+	service.EndpointSpec = &swarm.EndpointSpec{
+		Mode:  swarm.ResolutionModeVIP,
+		Ports: exposedPorts,
+	}
+
+	//Start the service
+	resp, err := d.Dockerd.ServiceCreate(context.Background(),
```

Comment:
could add a note or doc somewhere as to how this service is scaled/replicated across nodes or within a node. How volumes are accessed across nodes. How is a service instance mapped to a volume as each volume gets mounted (gref 0 -> 1)